### ISSUE TITLE
TextWire cannot unmarshall array of !type when value was written with typedMarshallable()

### ISSUE THREAD

**Issue by gregallen** [OpenHFT/Chronicle-Wire#102]
in this method:
net.openhft.chronicle.wire.TextWire.TextValueIn#consumeType

`parseUntil(sb, StopCharTesters.COMMA_SPACE_STOP);`

consumes the trailing `]`

perhaps should be COMMA_SPACE_BRACKET_STOP

and stop for ']' and '}'

(guessing that map with values that were types could have same issue)

**Comment by dpisklov** [comment]
@gregallen Can you supply a test case or at least example? We have quite a few places where it works.

**Comment by gregallen** [comment]
```

import org.junit.Test;

import net.openhft.chronicle.bytes.Bytes;
import net.openhft.chronicle.wire.AbstractMarshallable;
import net.openhft.chronicle.wire.TextWire;
import net.openhft.chronicle.wire.Wire;
import net.openhft.chronicle.wire.WireType;
import net.openhft.chronicle.wire.Wires;

public class ArrayTest
{
    static class Person extends AbstractMarshallable
    {
        Class<?>[] classes = {Object.class, Object.class};
    }

    @Test
    public void shouldUnmarshalArrayOfType() throws Exception
    {
        final Bytes<?> bytes = Wires.acquireBytes();

        final Wire wire = WireType.TEXT.apply(bytes);
        final Person person = new Person();
        wire.getValueOut().typedMarshallable(person);

        final TextWire textWire = TextWire.from(bytes.toString());
        textWire.getValueIn().typedMarshallable();
    }
}
```

**Comment by gregallen** [comment]
fails with

```
java.lang.IllegalStateException: Expected a ] was }


	at net.openhft.chronicle.wire.TextWire$TextValueIn.consumeSeq(TextWire.java:2410)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.consumeAny(TextWire.java:2361)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.consumeAny(TextWire.java:2387)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.consumeMap(TextWire.java:2419)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.consumeAny(TextWire.java:2357)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.readLengthMarshallable(TextWire.java:2345)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.marshallable(TextWire.java:2998)
	at net.openhft.chronicle.wire.Wires.objectMap(Wires.java:439)
	at net.openhft.chronicle.wire.Wires.object0(Wires.java:503)
	at net.openhft.chronicle.wire.ValueIn.object(ValueIn.java:578)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.objectWithInferredType0(TextWire.java:3309)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.objectWithInferredType(TextWire.java:3282)
	at net.openhft.chronicle.wire.TextWire$TextValueIn.typedMarshallable(TextWire.java:3072)
	at ArrayTest.shouldUnmarshalArrayOfType(ArrayTest.java:29)
```

**Comment by dpisklov** [comment]
@gregallen I fixed this particular issue (although the issue was in printing rather than parser - there must be a space before ] and after [ according to the YAML spec. However I found (and committed) another test case which is not working so will leave the issue open for now.

**Comment by gregallen** [comment]
@dpisklov Great! thanks for that. Any chance you could cut a version with the fix?

**Comment by dpisklov** [comment]
@gregallen there should be a release later this week, please bear with us!

**Comment by dpisklov** [comment]
@gregallen chronicle-wire-1.16.17 has been released today

**Comment by dpisklov** [comment]
Created a new issue for the outstanding test case and closing this one

**Comment by gregallen** [comment]
thanks @dpisklov

**Comment by hft-team-city** [comment]
Released in Chronicle-Wire-1.16.18 (https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-1.16.18), BOM-1.16.170 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-1.16.170)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
